tests/virtctl/scp.go

```diff
@@ -86,8 +87,12 @@ var _ = Describe("[sig-compute][virtctl]SCP", decorators.SigCompute, func() {
 
 	DescribeTable("should copy a local file back and forth", func(copyFn func(string, string, bool)) {
 		By("injecting a SSH public key into a VMI")
-		vmi := libvmi.NewAlpineWithTestTooling(
-			libvmi.WithCloudInitNoCloudUserData(libssh.RenderUserDataWithKey(pub)))
```

Comment:
I think this looks fine as it is. The reason it is ignored is the problem.

Options should override previous options, that is a reasonable behavior.
We have factories that are used a lot and from time to time there are exceptions. For those exceptions I think it is fine to override.

The alternative given here is possible, but it has a strong cons: It assumes that the one who writes the test knows all these details of building a VM/VMI. The point is not to require that heavy knowledge.